exonum/src/runtime/types.rs

```diff
@@ -63,12 +91,39 @@ impl AnyTx {
     }
 }
 
+/// The artifact identifier is required by the runtime to construct service instances.
+/// In other words an artifact identifier means same as class name, and a specific service
+/// instance is the class instance.
+///
+/// In string representation, the artifact identifier is written as follows:
+///
+/// `{runtime_id}:{artifact_name}`, where `runtime_id` is well known [runtime identifier],
+/// and `artifact_name` is unique name of artifact.
+///
+/// Artifact name can contains only these characters: `a-zA-Z0-9` and one of `_-./`.
```

Comment:
Isn't it more restrictive than semver (e.g., https://semver.org/#spec-item-10)?

Reply:
Yes, but I think this is normal because each change in artifact name produces a new unique artifact id.